tick/stateful/eval_function_node.go

```diff
@@ -171,10 +192,22 @@ func (n *EvalFunctionNode) EvalBool(scope *Scope, executionState ExecutionState)
 	return false, ErrTypeGuardFailed{RequestedType: ast.TBool, ActualType: ast.TypeOf(refValue)}
 }
 
+func (n *EvalFunctionNode) EvalMissing(scope *Scope, executionState ExecutionState) (*ast.Missing, error) {
+	refValue, err := n.callFunction(scope, executionState)
+	if err != nil {
+		return nil, err
+	}
+
+	if missingValue, isMissing := refValue.(*ast.Missing); isMissing {
+		return missingValue, nil
+	}
+	return nil, ErrTypeGuardFailed{RequestedType: ast.TMissing, ActualType: ast.TypeOf(refValue)}
+}
+
 // eval - generic evaluation until we have reflection/introspection capabillities so we can know the type of args
 // and return type, we can remove this entirely
 func eval(n NodeEvaluator, scope *Scope, executionState ExecutionState) (interface{}, error) {
-	retType, err := n.Type(scope, CreateExecutionState())
+	retType, err := n.Type(scope, executionState)
```

Comment:
Nothing should actually use Execution state, but that might not be true. I haven't tried to remove it again. I'll switch back to using  `CreateExecutionState()` until we rip out execution state from `Type`.